test/ready/shared.dart

```diff
@@ -2,97 +2,61 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-import 'package:scheduled_test/scheduled_test.dart';
+import 'package:test/test.dart';
 
 import '../utils.dart';
 
 void sharedTests() {
-  test('ready does not complete until after subscription', () {
-    var watcher = createWatcher(waitForReady: false);
+  test('ready does not complete until after subscription', () async {
+    var watcher = createWatcher();
 
     var ready = false;
     watcher.ready.then((_) {
       ready = true;
     });
+    await pumpEventQueue();
 
-    // Should not be ready yet.
-    schedule(() {
-      expect(ready, isFalse);
-    });
+    expect(ready, isFalse);
 
     // Subscribe to the events.
-    schedule(() {
-      var subscription = watcher.events.listen((event) {});
+    watcher.events.listen((event) {});
 
-      currentSchedule.onComplete.schedule(() {
-        subscription.cancel();
-      });
-    });
+    await expectLater(watcher.ready, completes);
```

Comment:
This can just be `await watcher.ready`. The `expectLater()` isn't doing anything here, since awaiting the future already guarantees the test won't complete until the future does.